bugzilla2gitlab/models.py

```diff
@@ -262,17 +268,29 @@ def parse_filename(cls, comment):
     def save(self):
         url = "{}/attachment.cgi?id={}".format(conf.bugzilla_base_url, self.id)
         result = _perform_request(url, "get", json=False)
+        filename = self.file_description
+        # Use real filename to store attachment but descriptive name for issue text
+        if 'Content-disposition' in result.headers:
+            regex = "^.*; filename=\"(.*)\"$"
+            matches = re.match(regex, result.headers['Content-disposition'], flags=re.M)
+            if matches:
+                filename = matches.group(1)
 
         url = "{}/projects/{}/uploads".format(conf.gitlab_base_url, conf.gitlab_project_id)
-        f = {"file": (self.filename, result.content)}
+        f = {"file": (filename, result.content)}
 
         attachment = _perform_request(url, "post", headers=self.headers, files=f, json=True,
                                       dry_run=conf.dry_run)
 
-        if conf.dry_run:
-            return "[attachment]({})".format(self.filename)
+        link = filename
```

Comment:
Same comment as above. This pattern doesn't seem appropriate for this situation. This renaming is also generally confusing.

Reply:
Please review the refactored version.